vizro-core/tests/unit/vizro/actions/_action_loop/test_get_action_loop_components.py

```diff
@@ -0,0 +1,168 @@
+from vizro.managers import model_manager
+
+
+@pytest.fixture
+def fundamental_components():
+    return [
+        dcc.Store(id="action_finished"),
+        dcc.Store(id="remaining_actions", data=[]),
+        html.Div(id="cycle_breaker_div", style={"display": "hidden"}),
+        dcc.Store(id="cycle_breaker_empty_output_store"),
+    ]
+
+
+@pytest.fixture
+def gateway_components(request):
+    components = request.param
+    actions_chain_ids = [model_manager[component].actions[0].id for component in components]
+    return [
+        dcc.Store(
+            id={"type": "gateway_input", "trigger_id": actions_chain_id},
+            data=f"{actions_chain_id}",
+        )
+        for actions_chain_id in actions_chain_ids
+    ]
+
+
+@pytest.fixture
+def action_trigger_components(request):
+    components = request.param
+    actions_ids = [model_manager[component].actions[0].actions[0].id for component in components]
+    return [dcc.Store(id={"type": "action_trigger", "action_name": action_id}) for action_id in actions_ids]
+
+
+@pytest.fixture
+def action_trigger_actions_id_component(request):
+    components = request.param
+    actions_ids = [model_manager[component].actions[0].actions[0].id for component in components]
+    return dcc.Store(
+        id="action_trigger_actions_id",
+        data=actions_ids,
+    )
+
+
+@pytest.fixture
+def trigger_to_actions_chain_mapper_component(request):
+    components = request.param
+    actions_chain_ids = [model_manager[component].actions[0].id for component in components]
+    return dcc.Store(
+        id="trigger_to_actions_chain_mapper",
+        data={
+            actions_chain_id: [action.id for action in model_manager[actions_chain_id].actions]
+            for actions_chain_id in actions_chain_ids
+        },
+    )
+
+
+@pytest.fixture
+def managers_one_page_two_components_two_controls():
+    """Instantiates managers with one page that contains two controls and two components."""
+    page = vm.Page(
+        id="test_page",
+        title="First page",
+        components=[
+            vm.Graph(
+                id="scatter_chart",
+                figure=px.scatter(px.data.gapminder(), x="lifeExp", y="gdpPercap"),
+            ),
+            vm.Button(
+                id="export_data_button",
+                actions=[vm.Action(id="export_data_action", function=export_data())],
+            ),
+        ],
+        controls=[
+            vm.Filter(id="filter_continent", column="continent", selector=vm.Dropdown(id="filter_continent_selector")),
+            vm.Parameter(
+                id="parameter_x",
+                targets=["scatter_chart.x"],
+                selector=vm.Dropdown(
+                    id="parameter_x_selector",
+                    options=["lifeExp", "gdpPercap", "pop"],
+                ),
+            ),
+        ],
+    )
+    # TODO: Call the Dashboard._pre_build() method once the pages registration is moved into this method.
```

Comment:
Same below

Reply:
Since these tests require all page components (and controls) to be pre-built, `Vizro._pre_build()` did his job (as it call `pre_build` for all models in the model_manager).

Reply:
Since these tests require all page components (and controls) to be pre-built, `Vizro._pre_build()` did the job (as `pre_build` method is called for all models in the model_manager). When the page registration was the part of the `Dashboard.build()` we needed to call `Vizro.build()` method to to establish page registration.